config/rbac/role.yaml

```diff
@@ -5,6 +5,18 @@ metadata:
   creationTimestamp: null
   name: odh-model-controller-role
 rules:
+- apiGroups:
+  - ""
+  resources:
+  - serviceaccounts
+  verbs:
+  - create
+  - get
+  - list
+  - patch
+  - update
+  - watch
+  - delete
```

Comment:
if the error is on the deletion process, why adding all these roles?

Reply:
no.. I just `delete` role for ServiceAccount. Earlier ServiceAccount is define with our resources in below group
```
- apiGroups:
  - ""
  resources:
  - configmaps
  - endpoints
  - namespaces
  - pods
  - serviceaccounts
  - secrets
  - services
  verbs:
  - create
  - get
  - list
  - patch
  - update
  - watch
  ```
  
  This above group delete role is missing for `ServiceAccount`. Now I moved out the ServiceAccount into seperate group and add delete role into it.
```
- apiGroups:
  - ""
  resources:
  - serviceaccounts
  verbs:
  - create
  - get
  - list
  - patch
  - update
  - watch
  - delete
  
- apiGroups:
  - ""
  resources:
  - configmaps
  - endpoints
  - namespaces
  - pods
  - secrets
  - services
  verbs:
  - create
  - get
  - list
  - patch
  - update
  - watch
  ```
  
  At the end I just add delete role for ServiceAccount. Everything else is same.